client/customer-effort-score-tracks/customer-effort-score-tracks.js

```diff
@@ -54,18 +54,42 @@ function CustomerEffortScoreTracks( {
 		return null;
 	}
 
-	if ( cesShownForActions.indexOf( action ) !== -1 && ! shown ) {
+	// We only want to return null early if the modal was already shown
+	// for this action *before* this component was initially instantiated.
+	//
+	// We want to make sure we still render CustomerEffortScore below
+	// (we don't want to return null early), if the modal was shown for this
+	// instantiation, so that the component doesn't go away while we are
+	// still showing it.
+	if ( cesShownForActions.indexOf( action ) !== -1 && ! modalShown ) {
 		return null;
 	}
 
-	const openedCallback = () => {
-		// Use the `shown` state value to only update the shown_for_actions
-		// option once.
-		if ( shown ) {
-			return;
-		}
+	const onNoticeShown = () => {
+		recordEvent( 'ces_snackbar_view', {
+			action,
+			store_age: storeAge,
+			...trackProps,
+		} );
+	};
+
+	const onNoticeDismissed = () => {
```

Comment:
Should this be triggered when clicking the entire modal (which dismisses it)?  I'm not seeing the event being called when this happens.

Reply:
No, this should only be triggered when the notice/snackbar is dismissed via the "X", which has not yet been merged (see #5623)

Reply:
This has been merged now